deltacat/compute/compactor_v2/deletes/utils.py

```diff
@@ -0,0 +1,160 @@
+    ] = [
+        (is_delete, sorted(list(delete_delta_group), key=lambda x: x.stream_position))
+        for (is_delete, _), delete_delta_group in itertools.groupby(
+            input_deltas, lambda d: (d.type is DeltaType.DELETE, d.delete_parameters)
+        )
+    ]
+    for (
+        is_delete_delta_sequence,
+        delete_delta_sequence,
+    ) in delete_deltas_sequence_grouped_by_delete_parameters:
+        if not is_delete_delta_sequence:
+            continue
+        starting_stream_position_of_delete_sequence: int = delete_delta_sequence[
+            0
+        ].stream_position
+        start_stream_spos_to_delete_delta_sequence[
+            starting_stream_position_of_delete_sequence
+        ] = delete_delta_sequence
+    return start_stream_spos_to_delete_delta_sequence
+
+
+def _get_delete_file_envelopes(
+    params: CompactPartitionParams,
+    delete_spos_to_delete_deltas: Dict[int, List],
+) -> List[DeleteFileEnvelope]:
+    """
+    Create a list of DeleteFileEnvelope objects from the given dictionary of delete deltas.
+    Args:
+        params (CompactPartitionParams): compaction session parameters
+        delete_spos_to_delete_deltas (Dict[int, List]): A dictionary where the keys are the stream positions of the
+            earliest delta in each group of consecutive DELETE deltas with the same delete parameters,
+            and the values are lists containing those deltas.
+    Returns:
+        List[DeleteFileEnvelope]: A list of DeleteFileEnvelope objects.
+    """
+    delete_file_envelopes = []
+    for (
+        start_stream_position,
+        delete_delta_sequence,
+    ) in delete_spos_to_delete_deltas.items():
+        consecutive_delete_tables: List[pa.Table] = []
+        for delete_delta in delete_delta_sequence:
+            assert (
+                delete_delta.delete_parameters is not None
+            ), "Delete type deltas are required to have delete parameters defined"
+            delete_columns: Optional[
+                List[str]
+            ] = delete_delta.delete_parameters.equality_column_names
+            assert len(delete_columns) > 0, "At least 1 delete column is required"
+            # delete columns should exist in underlying table
+            delete_dataset = params.deltacat_storage.download_delta(
+                delete_delta,
+                file_reader_kwargs_provider=params.read_kwargs_provider,
+                columns=delete_columns,
+                storage_type=StorageType.LOCAL,
+                max_parallelism=1,
+                **params.deltacat_storage_kwargs,
+            )
+            consecutive_delete_tables.extend(delete_dataset)
+        delete_table: pa.Table = pa.concat_tables(consecutive_delete_tables)
+        delete_file_envelope: DeleteFileEnvelope = DeleteFileEnvelope.of(
+            start_stream_position,
+            delta_type=DeltaType.DELETE,
+            table=delete_table,
+            delete_columns=delete_columns,
+        )
```

Comment:
Would we also need a way to override `table_storage_strategy` based on what is specified in `params`?